frontend/src/features/admin-form/create/builder-and-design/MagicFormBuilderModal/MagicFormBuilderModalContent/PromptDetailsScreen.tsx

```diff
@@ -0,0 +1,170 @@
+import { useState } from 'react'
+import { BiBulb, BiRightArrowAlt } from 'react-icons/bi'
+import {
+  Box,
+  Container,
+  Flex,
+  FormControl,
+  FormErrorIcon,
+  FormErrorMessage,
+  FormLabel,
+  ModalBody,
+  ModalHeader,
+  Text,
+} from '@chakra-ui/react'
+
+import { BxsErrorCircle } from '~assets/icons'
+import Button from '~components/Button'
+import Textarea from '~components/Textarea'
+
+import { useMagicFormBuilderWizard } from '../MagicFormBuilderWizardContext'
+
+export const MagicFormBuilderPromptDetailsScreen = (): JSX.Element => {
+  const {
+    handleDetailsSubmit,
+    formMethods,
+    isLoading,
+    isFetching,
+    handleBack,
+  } = useMagicFormBuilderWizard()
+
+  const {
+    register,
+    setValue,
+    formState: { errors },
+  } = formMethods
+
+  const handleUseIdea = (idea: string) => () => setValue('prompt', idea)
+  const [inputValue, setInputValue] = useState('')
+
+  return (
+    <>
+      <ModalHeader color="secondary.700">
+        <Container maxW={'42.5rem'} p={0}>
+          Write a prompt
+        </Container>
+      </ModalHeader>
+      <ModalBody whiteSpace="pre-wrap">
+        <Container maxW={'42.5rem'} p={0}>
+          <FormLabel>I want to create a form that collects...</FormLabel>
+          <FormControl isRequired isInvalid={!!errors.prompt}>
+            <Textarea
+              {...register('prompt', {
+                required: 'Please enter a prompt.',
+                onChange: (e) => setInputValue(e.target.value),
+                maxLength: {
+                  value: 300,
+                  message: 'Please enter at most 300 characters.',
+                },
+              })}
+              placeholder="personal particulars for an event"
+            />
+            <FormErrorMessage>
+              <FormErrorIcon h="1.5rem" as={BxsErrorCircle} />
+              {errors.prompt?.message} {inputValue.length}/300
```

Comment:
this 300 also seems like a magic number - can we codify it and inject it as a variable instead?